### PR TITLE
Add reporing of file length

### PR DESCRIPTION
**Pull request by sdhand**
Slight rounding issues. (Reported length usually tends to be shorter than the max length you can seek to), but not sure how to fix these.

### CODE REVIEW

**Comment by DinCahill** [comment]
yeah boi

File: src/audio/audio.hpp [hunk 1]
```diff
@@ -130,6 +132,7 @@ class NoAudio : public Audio
 	void SetPlaying(bool playing) override;
 	void SetPosition(std::uint64_t position) override;
 	std::uint64_t Position() const override;
+    std::uint64_t Length() const override;
```
Comment (MattWindsor91) [inline src/audio/audio.hpp]: Seems to be a bit of a space error here?

File: src/audio/audio.hpp [hunk 2]
```diff
@@ -165,6 +168,8 @@ class PipeAudio : public Audio
 	void SetPosition(std::uint64_t position) override;
 	std::uint64_t Position() const override;
 
+    std::uint64_t Length() const override;
```
Comment (MattWindsor91) [inline src/audio/audio.hpp]: And here

File: src/audio/audio_source.cpp [hunk 3]
```diff
@@ -34,7 +35,7 @@ std::uint64_t AudioSource::SamplesFromMicros(std::uint64_t micros) const
 	// need to convert the position to seconds then multiply by the rate.
 	// We do things in a slightly peculiar order to minimise rounding.
 
-	return (micros * this->SampleRate()) / 1000000;
+    return (micros * this->SampleRate()) / 1000000;
```
Comment (MattWindsor91) [inline src/audio/audio_source.cpp]: Huh?

File: src/audio/audio_source.cpp [hunk 4]
```diff
@@ -7,6 +7,7 @@
  * @see audio/audio_source.hpp
  */
 
+#include <cmath>
```
Comment (MattWindsor91) [inline src/audio/audio_source.cpp]: Is this needed on certain platforms?

**Comment by DinCahill** [comment]
`#include <cmath>` not needed on Linux, MinGW, or MSVC. @MattWindsor91 try a Mac build Kappa.

**Comment by sdhand** [comment]
cmath isn't needed on any system, I added it to try some floating point nonsense to make the rounding a bit better (it didn't work) and I forgot to remove it, sorry.

File: src/audio/audio_source.cpp (hunk 3 above)
Comment (sdhand) [inline src/audio/audio_source.cpp]: Sorry, blame my shitty editor setup on my laptop.

**Comment by MattWindsor91** [comment]
Aside from the space weirdness and having not yet tested it on macos (weekend project?), this looks like a go-er.

**Comment by MattWindsor91** [comment]
I've started a [pull request](https://github.com/UniversityRadioYork/baps3-spec/pull/20) for baps3-spec to bring the spec in line with this change.

**Comment by DinCahill** [comment]
Oh right there's more whitespace issues. I'll look at that.

**Comment by sdhand** [comment]
Sorry.... I should really fix my editor config on my laptop.

**Pull request merged**